apps/yapms/src/routes/app/[country]/[map]/[year]/initialize/LoadRegions.ts

```diff
@@ -179,61 +106,36 @@ function loadRegions(node: HTMLDivElement): void {
 			}
 		};
 
-		if (!newRegion.permaLocked) {
-			//If the region is marked as permaLocked, skip adding functionality
-			newRegion.nodes.region.onpointerdown = () => {
-				const currentMode = get(ModeStore);
-				switch (currentMode) {
-					case 'fill':
-						fillRegion(newRegion.id, true);
-						break;
-					case 'split':
-						splitRegion(newRegion.id);
-						break;
-					case 'edit':
-						editRegion(newRegion.id);
-						break;
-					case 'disable':
-						disableRegion(newRegion.id);
-						break;
-					case 'lock':
-						lockRegion(newRegion.id);
-						break;
-				}
-			};
-
-			newRegion.nodes.region.onmousemove = () => {
-				const currentMode = get(ModeStore);
-				const currentInteractions = get(InteractionStore);
-
-				if (currentMode === 'fill' && currentInteractions.has('KeyF')) {
-					fillRegion(newRegion.id, false);
-				}
-			};
-
-			if (newRegion.nodes.button !== null) {
-				newRegion.nodes.button.onpointerdown = newRegion.nodes.region.onpointerdown;
-				newRegion.nodes.button.onmousemove = newRegion.nodes.region.onmousemove;
-			}
-		}
-
 		newRegion.nodes.region.style.fill = tossupCandidate.margins[0].color;
 
-		if (newRegion.nodes.button) {
+		if (newRegion.nodes.button !== null) {
 			newRegion.nodes.button.style.fill = tossupCandidate.margins[0].color;
 		}
 
-		if (newRegion.nodes.text) {
+		if (newRegion.nodes.text !== null) {
 			const bottom = newRegion.nodes.text.querySelector('.bottom');
-			if (bottom) {
+			if (bottom !== null) {
 				bottom.textContent = newRegion.value.toString();
 			}
 		}
 
 		regionsForStore.push(newRegion);
-	});
+	}
 
 	RegionsStore.set(regionsForStore);
 }
 
-export default loadRegions;
+export function loadRegionsForApp(node: HTMLDivElement): void {
+	loadCandidateData(node);
+	setDefaultMode(node);
+	setCursorStyle(node);
+	createRegionStore(node);
+	setTransitionStyle(node);
+	setPointerEvents();
+}
+
+export function loadRegionsForView(node: HTMLDivElement): void {
+	loadCandidateData(node);
+	setDefaultMode(node);
```

Comment:
I don't think we need setDefaultMode here

Reply:
Fixed